src/library/scala/collection/mutable/HashMap.scala

```diff
@@ -358,7 +360,10 @@ class HashMap[K, V](initialCapacity: Int, loadFactor: Double)
     s.asInstanceOf[S with EfficientSplit]
   }
 
-  private[this] def growTable(newlen: Int) = {
+  private[this] def growTable(newlenLong: Long) = {
+    if (newlenLong > Int.MaxValue)
+      throw new BufferOverflowException()
```

Comment:
I'm not sure exactly what exception should be thrown, but I don't think it should be this one, which has to do with I/O. Perhaps an `IllegalStateException` with the message `"new HashMap table size exceeds maximum"`, or perhaps just a `RuntimeException`.

Reply:
Yeah, right. Done.